pkg/volume/awsebs/aws_util.go

```diff
@@ -125,6 +127,30 @@ func (util *AWSDiskUtil) CreateVolume(c *awsElasticBlockStoreProvisioner, node *
 	return name, volumeOptions.CapacityGB, labels, fstype, nil
 }
 
+// getCandidateZones finds possible zones that a volume can be created in
+func getCandidateZones(cloud *aws.Cloud, selectedNode *v1.Node, pvcName string) (sets.String, error) {
+	var err error
+	zones := sets.NewString()
+	if selectedNode != nil {
+		// For topology aware volume provisioning, node is already selected so we use the zone from
+		// selected node directly. We can assume the information is always available as node controller
+		// shall maintain it
+		if zone, ok := selectedNode.Labels[v1.LabelZoneFailureDomain]; ok {
+			klog.V(4).Infof("Using zone %q of selected node %q for PVC %q", zone, selectedNode.Name, pvcName)
+			zones.Insert(zone)
```

Comment:
We could just return completely empty set from here. There is no need to read `LabelZoneFailureDomain` too, because underlying code is already doing that.

Reply:
you are right. thanks for pointing out. Done.